actionpack/lib/action_dispatch/testing/integration.rb

```diff
@@ -408,17 +408,23 @@ def reset!
       end
 
       def create_session(app)
-        klass = APP_SESSIONS[app] ||= Class.new(Integration::Session) {
-          # If the app is a Rails app, make url_helpers available on the session
-          # This makes app.url_for and app.foo_path available in the console
-          if app.respond_to?(:routes)
-            include app.routes.url_helpers
-            include app.routes.mounted_helpers
-          end
-        }
+        klass = APP_SESSIONS[app] ||= Class.new(Integration::Session)
+        if app.respond_to?(:routes)
+          add_helpers(klass, app)
+        end
+
         klass.new(app)
       end
 
+      def add_helpers(session_class, app)
+      # If the app is a Rails app, make url_helpers available on the session
+      # This makes app.url_for and app.foo_path available in the console
+        session_class.instance_eval do
```

Comment:
Yeah, I'm aware of how to do it. I'm just not sure which to go with. As I
stated, I don't mind either way, but two core contributors have told me
they prefer two different ways.

So I'm curious, who cares more?
On Feb 14, 2015 1:48 PM, "Kasper Timm Hansen" <notifications@github.com>
wrote:

> In actionpack/lib/action_dispatch/testing/integration.rb
> <https://github.com/rails/rails/pull/18920#discussion_r24717613>:
>
> >          klass.new(app)
> >        end
> >
> > +      def add_helpers(session_class, app)
> > +      # If the app is a Rails app, make url_helpers available on the session
> > +      # This makes app.url_for and app.foo_path available in the console
> > +        session_class.instance_eval do
>
> @Spaceghost <https://github.com/Spaceghost> we're saying it's perfectly
> fine to do:
>
> session_class.include app.routes.url_helpers
> session_class.include app.routes.mounted_helpers
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/rails/rails/pull/18920/files#r24717613>.
>